File: design-patterns/cpp/facade/testing_framework.cpp
#include "testing_framework.h"

bool TestingFramework::runUnitTests() {
    std::cout << "Testing: Running unit tests..." << std::endl;
    simulateDelay(1500);
    std::cout << "Testing: Unit tests passed." << std::endl;
    return true;
}

bool TestingFramework::runIntegrationTests() {
    std::cout << "Testing: Running integration tests..." << std::endl;
    simulateDelay(3000);
    std::cout << "Testing: Integration tests passed." << std::endl;
    return true;
}

void TestingFramework::simulateDelay(int ms) {
    std::this_thread::sleep_for(std::chrono::milliseconds(ms));
} 

File: design-patterns/python/facade/testing_framework.py
import time

class TestingFramework:
    def run_unit_tests(self) -> bool:
        print("Testing: Running unit tests...")
        self._simulate_delay(1.5)
        print("Testing: Unit tests passed.")
        return True

    def run_integration_tests(self) -> bool:
        print("Testing: Running integration tests...")
        self._simulate_delay(3)
        print("Testing: Integration tests passed.")
        return True

    def _simulate_delay(self, seconds: int) -> None:
        time.sleep(seconds) 